### PR TITLE
feat: Provide option in phone input widget to turn off auto formatting

### LINKED ISSUE #10806: [Feature]: [Enhancement] Provide option to opt out of phone number formatting in phone Input widget
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

1. provide an option in the phone Input widget to turn off `as you type` formatting.
2. When the above-mentioned option is turned on, widget.text should be formatted while widget.value should be an unformatted phone number.

### Why should this be worked on?

Gives user more options to deal with the typed value

### PR DESCRIPTION
**Pull request by sbalaji1192**
## Description

Fixes #10806 

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce.
> Please also list any relevant details for your test configuration.

- Test A
- Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: feat/phone-input-formatting 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 55.06 **(0)** | 36.39 **(0)** | 34.71 **(0.01)** | 55.43 **(0)**
 :green_circle: | app/client/src/utils/DSLMigrations.ts | 73.27 **(0.15)** | 69.33 **(-0.06)** | 71.13 **(0)** | 73.08 **(0.16)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.71 **(-0.23)** | 40.83 **(-0.84)** | 36.21 **(0)** | 56.74 **(-0.25)**
 :sparkles: :new: | **app/client/src/utils/migrations/PhoneInputWidgetMigrations.ts** | **100** | **90** | **100** | **100**
 :red_circle: | app/client/src/widgets/PhoneInputWidget/widget/index.tsx | 40.63 **(-5.15)** | 6.52 **(-1.81)** | 42.86 **(-3.29)** | 40.43 **(-5.25)**</details>

### CODE REVIEW

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/FormWidgets/Phone_input_spec.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ describe("Phone input widget - ", () => {
 
   it("should check for the format and dialCode", () => {
     cy.get(`.t--widget-${widgetName} input`).clear();
-    cy.wait(300);
+    cy.wait(500);
```
Comment (sbalaji1192) [inline app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/FormWidgets/Phone_input_spec.js:25]: increasing wait time to fix flakiness

**Comment by shwetha-ramesh** [comment]
@sbalaji1192 tested this PR. DP looks good
image

Moving to Done

File: app/client/src/widgets/PhoneInputWidget/widget/index.tsx [hunk 2]
```diff
@@ -121,26 +135,40 @@ class PhoneInputWidget extends BaseInputWidget<
   }
 
   static getDerivedPropertiesMap(): DerivedPropertiesMap {
-    return _.merge(super.getDerivedPropertiesMap(), {
```
Comment (jsartisan) [inline app/client/src/widgets/PhoneInputWidget/widget/index.tsx:124]: Any reason to remove the merge?

**Review by jsartisan — APPROVED** [review]

**Review by jsartisan — APPROVED** [review]

**Comment by shwetha-ramesh** [comment]
@sbalaji1192 Retested this PR for #10806 & #10859. Everything is working good. Tested for migration too. Nothing seems to break.
Moving to Done.

**Review by jsartisan — APPROVED** [review]

**Pull request merged**